python/ray/air/util/torch_dist.py

```diff
@@ -0,0 +1,158 @@
+"""This file is modeled after ray/python/ray/train/torch/config.py
+
+The logics are duplicated right now to allow maximum flexibility for
+setting up PyTorch DDP process groups outside the context of Ray Train.
+Eventually, these use cases should be consolidated.
+"""
+
+from abc import ABC
+from datetime import timedelta
+import os
+import torch.distributed as dist
+from typing import Callable, List, T
+
+import ray
+from ray.actor import ActorHandle
+from ray.air._internal.util import skip_exceptions, exception_cause
+from ray.train._internal.utils import get_address_and_port
+from ray.train.constants import DEFAULT_NCCL_SOCKET_IFNAME
+
+
+class TorchDistributedWorker(ABC):
+    """Defines the interfaces required by the `init_torch_dist_process_group()`.
+
+    This is modeled after RayTrainerWorker, which allows arbitrary functions
+    to be executed on a remote DDP worker.
+    """
+
+    def execute(self, func: Callable[..., T], *args, **kwargs) -> T:
+        """Executes the input function and returns the output.
+
+        Args:
+            func: The function to execute.
+            args, kwargs: The arguments to pass into func.
+        """
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            skipped = skip_exceptions(e)
+            raise skipped from exception_cause(skipped)
```

Comment:
we want to remove all this skip_exceptions stuff? It only works when used with Train/Tune

Reply:
ah, good to know. thanks for the comment.